azurerm/internal/services/costmanagement/tests/resource_arm_cost_management_export_resource_group_test.go

```diff
@@ -0,0 +1,197 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMCostManagementExportResourceGroup_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCostManagementExportResourceGroupExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMCostManagementExportResourceGroup_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMCostManagementExportResourceGroupExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMCostManagementExportResourceGroupExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).CostManagement.ExportClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		id, err := parse.CostManagementExportResourceGroupID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Get(ctx, id.ResourceId, id.Name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on costManagementExportResourceGroupClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Cost Management Export ResourceGroup %q (resource group: %q) does not exist", id.Name, id.ResourceId)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMCostManagementExportResourceGroupDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).CostManagement.ExportClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_cost_management_export_resource_group" {
+			continue
+		}
+
+		id, err := parse.CostManagementExportResourceGroupID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		resp, err := client.Get(ctx, id.ResourceId, id.Name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Cost Management Export ResourceGroup still exists: %q", id.Name)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMCostManagementExportResourceGroup_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-cmerg-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                = "unlikely23exst2acct%s"
+  resource_group_name = azurerm_resource_group.test.name
+
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_cost_management_export_resource_group" "test" {
+  name                    = "accrg%d"
```

Comment:
Should we use some capitals here?

Reply:
Not this time. It only accepts lower case